src/models.ts

```diff
@@ -0,0 +1,58 @@
+export class Book {
+  id: string;
+  title: string;
+  author: string;
+  isLent: boolean;
+  currentLendeeId?: string;
+  dueDate?: string;
+  lendingHistory: LendingRecord[];
+
+  constructor(
+    id: string,
+    title: string,
+    author: string,
+    isLent: boolean = false,
+    currentLendeeId?: string,
+    dueDate?: string,
+    lendingHistory: LendingRecord[] = []
+  ) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+    this.isLent = isLent;
+    this.currentLendeeId = currentLendeeId;
+    this.dueDate = dueDate;
+    this.lendingHistory = lendingHistory;
+  }
+}
+
+export class Customer {
+  id: string;
+  name: string;
+  email: string;
+
+  constructor(id: string, name: string, email: string) {
+    this.id = id;
+    this.name = name;
+    this.email = email;
+  }
+}
+
+export class LendingRecord {
```

Comment:
I think LendingRecord should also have an id, not going to use it for now, but all entries in most of the DBs have unique keys(id), and if we want to show those records somewhere in the UI, might be good to have the ID for CRUD operations or such

Reply:
I added it, your point makes sense